virtualizarr/manifests/manifest.py

```diff
@@ -283,12 +297,18 @@ def to_zarr_json(self, filepath: str) -> None:
             json.dump(entries, json_file, indent=4, separators=(", ", ": "))
 
     @classmethod
-    def _from_kerchunk_chunk_dict(cls, kerchunk_chunk_dict) -> "ChunkManifest":
-        chunkentries = {
-            cast(ChunkKey, k): ChunkEntry.from_kerchunk(v).dict()
-            for k, v in kerchunk_chunk_dict.items()
-        }
-        return ChunkManifest(entries=cast(ChunkDict, chunkentries))
+    def _from_kerchunk_chunk_dict(
+        cls,
+        kerchunk_chunk_dict: Dict[ChunkKey, str | tuple[str] | tuple[str, int, int]],
+    ) -> "ChunkManifest":
+        chunk_entries: dict[ChunkKey, ChunkDictEntry] = {}
+        for k, v in kerchunk_chunk_dict.items():
+            if isinstance(v, (str, bytes)):
+                raise NotImplementedError("TODO: handle inlined data")
+            elif not isinstance(v, (tuple, list)):
```

Comment:
Isn't this incorrect typing? You allow `v` to be a `list`, then pass the list into a method which only accepts `tuple`s (i.e. `ChunkEntry.from_kerchunk()`.

Reply:
I'm pulling some really sketchy tricks with typing here.

My rationale is that kerchunk can and probably will give us weird types, and we should be robust with handling them at runtime. While `ChunkEntry.from_kerchunk` expects `tuple`s it can still handle lists if need be, even though I don't tell the typing system.

From the type checking perspective, the instance check for `list` is vacuous since the type of `v` is `str | tuple[str] | tuple[str, int, int]`. So basically what I'm saying is, "I want to raise a type error if I don't get a `tuple`, but if I happen to find a `list` I'll allow it."

Do you find this an acceptable strategy, or would you like to change it?

Ideally we'd add types to kerchunk, but I don't have time for that right now.

Reply:
Okay! So it's intentional.

I think this is fine - it still deals fine with the types the hints claim it will, plus you're saying that it will also silently succeed on some other types too.

> Ideally we'd add types to kerchunk, but I don't have time for that right now.

Yep. In the long term the plan is to re-implement kerchunk's readers without going via the intermediate kerchunk reference format, at which point we will hopefully be able to remove all the code in `kerchunk.py` anyway.